core/federated/RTI/rti_local.c

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file
+ * @author Erling Jellum (erling.r.jellum@ntnu.no)
+ * @author Edward A. Lee (eal@berkeley.edu)
+ * @author Chadlia Jerad (chadlia.jerad@ensi-uma.tn)
+ * @author Soroush Bateni (soroush@utdallas.edu)
+ * @copyright (c) 2020-2023, The University of California at Berkeley
+ * License in [BSD 2-clause](https://github.com/lf-lang/reactor-c/blob/main/LICENSE.md)
+*/
+
+#ifdef LF_ENCLAVES
+#include "rti_local.h"
+#include "rti_common.h"
+#include "util.h"
+#include "platform.h"
+#include "environment.h"
+#include "trace.h"
+#include "reactor.h"
+
+
+// The following are macros for to be used in the enter and exit
+// of the local RTI API. We extract this to macros to make sure that 
+// mutexes are acquired and released in the correct order.
+#define LTC_LOCKED_PROLOGUE(enclave) \
+    do { \
+        lf_mutex_unlock(&enclave->env->mutex); \
+    } while(0)
+
+#define LTC_LOCKED_EPILOGUE(enclave) \
+    do { \
+        lf_mutex_lock(&enclave->env->mutex); \
+    } while(0)
+
+#define NET_LOCKED_PROLOGUE(enclave) \
```

Comment:
What is the value of `rti_next_event_tag_locked` requiring that the enclave mutex be held when it is called if it is going to immediately release that lock and acquire the global mutex?  Wouldn't it be better to just require that the global mutex be held?

Reply:
It is less about requiring it to hold the mutex and more about that calls to this function only appears within critical sections of the enclave. So thats why I moved the unlocking and locking of the enclave mutex inside the function. 

If I remember correctly, there isnt a way to grab the "global" mutex before calling the function. The global mutex is the RTI mutex and we only have it in scope inside the RTI files